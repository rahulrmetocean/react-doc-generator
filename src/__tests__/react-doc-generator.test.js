// https://travis-ci.org/marborkowski/react-doc-generator

require("@babel/polyfill");
const TEST_TIMEOUT = 120000;
jasmine.DEFAULT_TIMEOUT_INTERVAL = TEST_TIMEOUT; // eslint-disable-line no-undef

const path = require("path");
const fs = require("fs");
const spawn = require("child_process").spawn;

function run(command, args) {
  let stdout = [];
  let stderr = [];

  return new Promise((resolve, reject) => {
    let spawned = spawn(command, args);
    spawned.stdout.on("data", data => {
      stdout.push(data);
    });
    spawned.stderr.on("data", data => stderr.push(data));
    spawned.on("close", () => resolve([stdout.join(""), stderr.join("")]));
    spawned.on("error", err => {
      throw err;
    });
  }).catch(error => {
    console.log(error, "Error");
    throw error;
  });
}

function loadDoc(file) {
  return new Promise((resolve, reject) => {
    fs.readFile(file, "utf8", function(err, data) {
      if (err) {
        reject(err);
      } else {
        resolve(data);
      }
    });
  });
}
let binPath = path.join(__dirname, "../../dist/react-doc-generator.js");
  fs.chmodSync(binPath, "0777");
describe("react-doc-generator", () => {
  
  it("has the proper console output", async () => {
    try {
      const stdout = await run("node", [
        binPath,
        "src/__mocks__/basic",
        "-o",
        "./dist/DOCUMENTATION.md",
        "-d",
        "someduck",
        "-s",
        "someducksidebar",
      ]);
      const output = stdout[0];
    expect(output).toMatchSnapshot()
    } catch (e) {
      console.error(e);
      throw e;
    }
  });

  it("return the proper message when given extensions not found", async () => {
    try {
      const stdout = await run("node", [
        binPath,
        "src/__mocks__/basic",
        "-o",
        "./dist/DOCUMENTATION.md",
        "-x",
        "4hs0,kku4",
      ]);
      const output = stdout[0];
      expect(output).toMatchSnapshot()
    } catch (e) {
      throw e;
    }
  });

  it("contains help section if no argument is available in query", async () => {
    try {
      const stdout = await run("node", [
        binPath,
        "-o",
        "./dist/DOCUMENTATION.md",
        "-d",
        "someduck",
        "-s",
        "someducksidebar"
      ]);
      const output = stdout[0];
      expect(output).toMatchSnapshot()
    } catch (e) {
      throw e;
    }
  });
});
describe("output file", () => {
  it("has needed values", async () => {
    try {
      await run("node",[
        binPath,
        "src/__mocks__/basic",
        "-o",
        "./dist/basic.md",
      ]);
      const result = await loadDoc( "./dist/basic.md");
      const lines = result.split("\n");
      expect(lines).toMatchSnapshot()
    } catch (e) {
      throw e;
    }
  });
  it("Defaults docusaurus id and sidebar when values were not passed", async () => {
    try {
      await run("node",[
        binPath,
        "src/__mocks__/basic",
        "-o",
        "./dist/defaultsidebar.md",
      ]);
      const result = await loadDoc("./dist/defaultsidebar.md");
      const lines = result.split("\n");
      expect(lines).toMatchSnapshot()
    } catch (e) {
      throw e;
    }
  });
  it("Populates docusaurus id and sidebar when values were passed", async () => {
    try {
      await run("node",[
        binPath,
        "src/__mocks__/basic",
        "-o",
        "./dist/sidebar.md",
        "-d",
        "someduck",
        "-s",
        "someducksidebar"
      ]);
      const result = await loadDoc('./dist/sidebar.md');
      const lines = result.split("\n");
      expect(lines).toMatchSnapshot()
    } catch (e) {
      throw e;
    }
  });
  it("Works well for flow components as well", async () => {
    try {
      const stdout = await run("node",[
        binPath,
        "src/__mocks__/flow",
        "-o",
        "./dist/flow.md",
        "-d",
        "someduck",
        "-s",
        "someducksidebar"
      ]);
      console.log(stdout[0])
      const result = await loadDoc("./dist/flow.md");
      const lines = result.split("\n");
      expect(lines).toMatchSnapshot()
    } catch (e) {
      throw e;
    }
  });
});
