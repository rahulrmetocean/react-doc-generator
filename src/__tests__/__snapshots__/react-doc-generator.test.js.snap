// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`output file Defaults docusaurus id and sidebar when values were not passed 1`] = `
Array [
  "---
",
  "id: defaultId
",
  "title: Components
",
  "sidebar_label: defaultSidebarLabel
",
  "---
",
  "
",
  "Components
",
  "----------
",
  "
",
  "**C:\\\\Users\\\\rravindran\\\\metocean\\\\react-doc-generator\\\\src\\\\__mocks__\\\\basic\\\\ExampleComponent.js**
",
  "
",
  "### 1. General component description.
",
  "
",
  "fdgdfgdf gfdgfdg fdgfdgdfg
   ",
  "gdfgfdgdfg dfgdfgfdg dfg df getDefaultPropsg fdgfd
   ",
  "gfdgfdgdfgfdg.
   ",
  "   ",
  "   ",
  "Example:
   ",
  "\`\`\`html
   ",
  "<Component name=\\"bebe\\" />
   ",
  "\`\`\`   ",
  "",
  "
",
  "
",
  "
",
  "Property | Type | Required | Default value | Description
",
  ":--- | :--- | :--- | :--- | :---
",
  "toe|string|no|&#x27;&#x27;|Description of prop &quot;toe&quot; has one break line here following more comments and has default empty string.
",
  "finger|string|no|&#x27;medium&#x27;|Description of prop &quot;finger&quot;.
",
  "foo|number|no|42|Description of prop &quot;foo&quot;.
",
  "bar|custom|no|21|Description of prop &quot;bar&quot; (a custom validation function).
",
  "baz|union|no||
",
  "-----
",
  "**C:\\\\Users\\\\rravindran\\\\metocean\\\\react-doc-generator\\\\src\\\\__mocks__\\\\basic\\\\inside\\\\OtherExampleComponent.js**
",
  "
",
  "### 1. Component1
",
  "
",
  "null   ",
  "",
  "
",
  "
",
  "
",
  "Property | Type | Required | Default value | Description
",
  ":--- | :--- | :--- | :--- | :---
",
  "foo|number|no|10000099999|Description of prop &quot;foo&quot;.
",
  "bar|custom|no||Description of prop &quot;bar&quot; (a custom validation function).
",
  "baz|union|no||
",
  "onExit|func|no|See code|
",
  "### 2. DUPA
",
  "
",
  "Blah blah blah...
   ",
  "fdfdfsdf
   ",
  "fdsfsd   ",
  "",
  "
",
  "
",
  "
",
  "Property | Type | Required | Default value | Description
",
  ":--- | :--- | :--- | :--- | :---
",
  "foo|number|no||Description of prop &quot;foo&quot;.
",
  "bar|custom|no||Description of prop &quot;bar&quot; (a custom validation function).
",
  "baz|union|no||
",
  "-----
",
  "
",
  "
",
  "",
]
`;

exports[`output file Populates docusaurus id and sidebar when values were passed 1`] = `
Array [
  "---
",
  "id: someduck
",
  "title: Components
",
  "sidebar_label: someducksidebar
",
  "---
",
  "
",
  "Components
",
  "----------
",
  "
",
  "**C:\\\\Users\\\\rravindran\\\\metocean\\\\react-doc-generator\\\\src\\\\__mocks__\\\\basic\\\\ExampleComponent.js**
",
  "
",
  "### 1. General component description.
",
  "
",
  "fdgdfgdf gfdgfdg fdgfdgdfg
   ",
  "gdfgfdgdfg dfgdfgfdg dfg df getDefaultPropsg fdgfd
   ",
  "gfdgfdgdfgfdg.
   ",
  "   ",
  "   ",
  "Example:
   ",
  "\`\`\`html
   ",
  "<Component name=\\"bebe\\" />
   ",
  "\`\`\`   ",
  "",
  "
",
  "
",
  "
",
  "Property | Type | Required | Default value | Description
",
  ":--- | :--- | :--- | :--- | :---
",
  "toe|string|no|&#x27;&#x27;|Description of prop &quot;toe&quot; has one break line here following more comments and has default empty string.
",
  "finger|string|no|&#x27;medium&#x27;|Description of prop &quot;finger&quot;.
",
  "foo|number|no|42|Description of prop &quot;foo&quot;.
",
  "bar|custom|no|21|Description of prop &quot;bar&quot; (a custom validation function).
",
  "baz|union|no||
",
  "-----
",
  "**C:\\\\Users\\\\rravindran\\\\metocean\\\\react-doc-generator\\\\src\\\\__mocks__\\\\basic\\\\inside\\\\OtherExampleComponent.js**
",
  "
",
  "### 1. Component1
",
  "
",
  "null   ",
  "",
  "
",
  "
",
  "
",
  "Property | Type | Required | Default value | Description
",
  ":--- | :--- | :--- | :--- | :---
",
  "foo|number|no|10000099999|Description of prop &quot;foo&quot;.
",
  "bar|custom|no||Description of prop &quot;bar&quot; (a custom validation function).
",
  "baz|union|no||
",
  "onExit|func|no|See code|
",
  "### 2. DUPA
",
  "
",
  "Blah blah blah...
   ",
  "fdfdfsdf
   ",
  "fdsfsd   ",
  "",
  "
",
  "
",
  "
",
  "Property | Type | Required | Default value | Description
",
  ":--- | :--- | :--- | :--- | :---
",
  "foo|number|no||Description of prop &quot;foo&quot;.
",
  "bar|custom|no||Description of prop &quot;bar&quot; (a custom validation function).
",
  "baz|union|no||
",
  "-----
",
  "
",
  "
",
  "",
]
`;

exports[`output file Works well for flow components as well 1`] = `
Array [
  "---
",
  "id: someduck
",
  "title: Components
",
  "sidebar_label: someducksidebar
",
  "---
",
  "
",
  "Components
",
  "----------
",
  "
",
  "**C:\\\\Users\\\\rravindran\\\\metocean\\\\react-doc-generator\\\\src\\\\__mocks__\\\\flow\\\\ComponentWithFlow.js**
",
  "
",
  "### 1. Component1
",
  "
",
  "null   ",
  "",
  "
",
  "
",
  "
",
  "Property | Type | Required | Default value | Description
",
  ":--- | :--- | :--- | :--- | :---
",
  "foo|number|no|10000099999|
",
  "bar|() &#x3D;&gt; void|yes||Description of prop &quot;bar&quot; (a custom validation function).
",
  "baz|number|string|yes||Description of a property with a flow or operation
",
  "someProp|{   [key:string]: string }|yes||Description for a property which is more than one line
",
  "onExit|() &#x3D;&gt; void|no|() &#x3D;&gt; {     console.debug(&#x27;onExit&#x27;); }|Handler to invoke a function
",
  "customObjectProp|{   firstName:string,   lastName:string }|yes||
",
  "### 2. DUPA
",
  "
",
  "Blah blah blah...
   ",
  "fdfdfsdf
   ",
  "fdsfsd   ",
  "",
  "
",
  "
",
  "
",
  "Property | Type | Required | Default value | Description
",
  ":--- | :--- | :--- | :--- | :---
",
  "foo|number|yes||Description of prop &quot;foo&quot;.
",
  "bar|() &#x3D;&gt; mixed|yes||Description of prop &quot;bar&quot; (a custom validation function).
",
  "baz|number | string|yes||
",
  "-----
",
  "
",
  "
",
  "",
]
`;

exports[`output file has needed values 1`] = `
Array [
  "---
",
  "id: defaultId
",
  "title: Components
",
  "sidebar_label: defaultSidebarLabel
",
  "---
",
  "
",
  "Components
",
  "----------
",
  "
",
  "**C:\\\\Users\\\\rravindran\\\\metocean\\\\react-doc-generator\\\\src\\\\__mocks__\\\\basic\\\\ExampleComponent.js**
",
  "
",
  "### 1. General component description.
",
  "
",
  "fdgdfgdf gfdgfdg fdgfdgdfg
   ",
  "gdfgfdgdfg dfgdfgfdg dfg df getDefaultPropsg fdgfd
   ",
  "gfdgfdgdfgfdg.
   ",
  "   ",
  "   ",
  "Example:
   ",
  "\`\`\`html
   ",
  "<Component name=\\"bebe\\" />
   ",
  "\`\`\`   ",
  "",
  "
",
  "
",
  "
",
  "Property | Type | Required | Default value | Description
",
  ":--- | :--- | :--- | :--- | :---
",
  "toe|string|no|&#x27;&#x27;|Description of prop &quot;toe&quot; has one break line here following more comments and has default empty string.
",
  "finger|string|no|&#x27;medium&#x27;|Description of prop &quot;finger&quot;.
",
  "foo|number|no|42|Description of prop &quot;foo&quot;.
",
  "bar|custom|no|21|Description of prop &quot;bar&quot; (a custom validation function).
",
  "baz|union|no||
",
  "-----
",
  "**C:\\\\Users\\\\rravindran\\\\metocean\\\\react-doc-generator\\\\src\\\\__mocks__\\\\basic\\\\inside\\\\OtherExampleComponent.js**
",
  "
",
  "### 1. Component1
",
  "
",
  "null   ",
  "",
  "
",
  "
",
  "
",
  "Property | Type | Required | Default value | Description
",
  ":--- | :--- | :--- | :--- | :---
",
  "foo|number|no|10000099999|Description of prop &quot;foo&quot;.
",
  "bar|custom|no||Description of prop &quot;bar&quot; (a custom validation function).
",
  "baz|union|no||
",
  "onExit|func|no|See code|
",
  "### 2. DUPA
",
  "
",
  "Blah blah blah...
   ",
  "fdfdfsdf
   ",
  "fdsfsd   ",
  "",
  "
",
  "
",
  "
",
  "Property | Type | Required | Default value | Description
",
  ":--- | :--- | :--- | :--- | :---
",
  "foo|number|no||Description of prop &quot;foo&quot;.
",
  "bar|custom|no||Description of prop &quot;bar&quot; (a custom validation function).
",
  "baz|union|no||
",
  "-----
",
  "
",
  "
",
  "",
]
`;

exports[`react-doc-generator contains help section if no argument is available in query 1`] = `
"[37m[39m

[37mREACT DOC GENERATOR v1.2.5[39m
[37mby Marcin Borkowski <marborkowski@gmail.com>[39m
[]
[31mPlease specify <dir> as the first argument![39m

  Usage: react-doc-generator <dir> [options]

  Options:

    -h, --help                        output usage information
    -V, --version                     output the version number
    -x, --extensions <items>          Include only these file extensions. Default: js,jsx
    -i, --ignore <items>              Folders to ignore. Default: node_modules,__tests__,__mocks__
    -e, --exclude-patterns <items>    Filename patterns to exclude. Default: []
    -t, --title [value]               Document title. Default: 'Components'
    -o, --output <file>               Markdown file to write. Default: 'DOCUMENTATION.MD'
    -d, --docusaurus-id <value>       Id for docusaurus file
    -s, --docusaurus-sidebar <value>  Sidebar label

"
`;

exports[`react-doc-generator has the proper console output 1`] = `
"[37m[39m

[37mREACT DOC GENERATOR v1.2.5[39m
[37mby Marcin Borkowski <marborkowski@gmail.com>[39m
C:\\\\Users\\\\rravindran\\\\metocean\\\\react-doc-generator\\\\src\\\\__mocks__\\\\basic\\\\ExampleComponent.js filename
C:\\\\Users\\\\rravindran\\\\metocean\\\\react-doc-generator\\\\src\\\\__mocks__\\\\basic\\\\inside\\\\OtherExampleComponent.js filename
C:\\\\Users\\\\rravindran\\\\metocean\\\\react-doc-generator\\\\src\\\\__mocks__\\\\basic\\\\ExampleComponent.js,1,[32mOK.[39m,C:\\\\Users\\\\rravindran\\\\metocean\\\\react-doc-generator\\\\src\\\\__mocks__\\\\basic\\\\inside\\\\OtherExampleComponent.js,2,[32mOK.[39m


┌──────────────────────────────────────────────────────────────────────────────────────────────────────┬────────────┬────────┐
│ [36mPath[39m                                                                                                 │ [36mComponents[39m │ [36mStatus[39m │
├──────────────────────────────────────────────────────────────────────────────────────────────────────┼────────────┼────────┤
│ C:\\\\Users\\\\rravindran\\\\metocean\\\\react-doc-generator\\\\src\\\\__mocks__\\\\basic\\\\ExampleComponent.js             │ 1          │ [32mOK.[39m    │
├──────────────────────────────────────────────────────────────────────────────────────────────────────┼────────────┼────────┤
│ C:\\\\Users\\\\rravindran\\\\metocean\\\\react-doc-generator\\\\src\\\\__mocks__\\\\basic\\\\inside\\\\OtherExampleComponent.js │ 2          │ [32mOK.[39m    │
└──────────────────────────────────────────────────────────────────────────────────────────────────────┴────────────┴────────┘
"
`;

exports[`react-doc-generator return the proper message when given extensions not found 1`] = `
"[37m[39m

[37mREACT DOC GENERATOR v1.2.5[39m
[37mby Marcin Borkowski <marborkowski@gmail.com>[39m
[1m[33mWarning:[39m[22m [33mCould not find any files matching the file type: \`*.4hs0\` OR \`*.kku4\`[39m

┌──────┬────────────┬────────┐
│ [36mPath[39m │ [36mComponents[39m │ [36mStatus[39m │
└──────┴────────────┴────────┘
"
`;
